src/main/kotlin/reposfinder/logic/ReposFinder.kt

```diff
@@ -0,0 +1,189 @@
+    private val logger: Logger
+
+    private val dumpDir = File(config.dumpDir)
+    val reposStorage: ReposStorage
+
+    private val jsonMapper = jacksonObjectMapper()
+
+    init {
+        dumpDir.mkdirs()
+        logger = Logger(config.logPath, isDebug = config.isDebug)
+        reposStorage = ReposStorage(config.urls, config.dumpDir, config.dumpEveryNRepos, logger = logger)
+        limits = RateLimits(config.isCore, config.isGraphQL, config.token, logger = logger)
+    }
+
+    override fun run() {
+        if (status != Status.READY) {
+            return
+        }
+        status = Status.WORKING
+        status = try {
+            logStartSummary()
+            limits.update()
+            logLimits()
+            searchRepos()
+            reposStorage.dump()
+            Status.DONE
+        } catch (e: Exception) {
+            logger.add("============== ERROR WHILE SEARCH RUNNING ==============")
+            logger.add(e.stackTraceToString())
+            Status.ERROR
+        } finally {
+            logEndSummary()
+            logger.dump()
+        }
+    }
+
+    fun interrupt() {
+        status = Status.INTERRUPTED
+    }
+
+    private fun searchRepos() {
+        if (limits.isNoLimits()) {
+            status = Status.BAD_TOKEN_LIMITS
+            logger.add("ZERO LIMITS ERROR (maybe bad token)")
+            return
+        }
+        for (repo in reposStorage.allRepos) {
+            waitingControl(limits.check())
+            if (status != Status.WORKING) {
+                logger.add("SEARCH WAS INTERRUPTED WITH STATUS $status")
+                break
+            }
+            logger.add("> processing: /${repo.owner}/${repo.name}")
+
+            val isCoreGood = repo.loadCore() && repo.isGood(config.coreFilters)
+            val isGraphQLGood = repo.loadGraphQL() && repo.isGood(config.graphQLFilters)
+
+            if (isCoreGood && isGraphQLGood) {
+                reposStorage.addGood(repo)
+            } else {
+                reposStorage.addBad(repo)
+            }
+        }
+    }
+
+    private fun Repository.loadCore(): Boolean {
+        var isGood = true
+        if (config.isCore) {
+            if (!config.isOnlyContributors) {
+                isGood = this.loadCore(jsonMapper, config.token) && isGood
+                limits.core.register()
+                Thread.sleep(config.sleepTimeBetweenRequests)
+            }
+            if (config.isContributors) {
+                isGood = this.loadContributors(config.isAnonContributors, config.token) && isGood
+                limits.core.register()
+                Thread.sleep(config.sleepTimeBetweenRequests)
+            }
+        }
+        return isGood
+    }
+
+    private fun Repository.loadGraphQL(): Boolean {
+        var isGood = true
+        if (config.isGraphQL) {
+            if (config.isCommitsCount) {
+                isGood = this.loadGraphQL(GraphQLQueries.COMMITS_COUNT, jsonMapper, config.token) && isGood
```

Comment:
Why you write `&& isGood`? Since you declared this value as true this part of an expression is unused.

Reply:
fixed